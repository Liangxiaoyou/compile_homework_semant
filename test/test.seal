/* 
this a correct example of SEAL 
*/
var main Int;
var a Void;
var a Bool;
func main() Int{
    x=1;
return ;
}
func main2() Int{
    return ;
    if (2 > 1) {var x Int;}
    if (x > 1) {var x Int;}
}
func main4() Void{
    return ;
}
