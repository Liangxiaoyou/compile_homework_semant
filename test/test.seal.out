#4
Program
  #4
  Variable Declaration
    #4
    Variable
      (name)
      a
      (type)
      Int
  #5
  Call Declaration
    (name)
    main
    (parameters)
    (
    )
    (return type)
    Void
    (body)
    #5
    Statement Block
      (variable declarations)
      (
      )
      (statements)
      (
      #6
      ReturnStmt
        (return value)
        #6
        No_expr
      )
  #8
  Call Declaration
    (name)
    main1
    (parameters)
    (
    )
    (return type)
    Void
    (body)
    #8
    Statement Block
      (variable declarations)
      (
      )
      (statements)
      (
      #9
      ReturnStmt
        (return value)
        #9
        No_expr
      )
